beacon-chain/rpc/beaconv1/state.go

```diff
@@ -231,6 +237,9 @@ func (bs *Server) finalizedStateRoot(ctx context.Context) ([]byte, error) {
 	if err != nil {
 		return nil, status.Errorf(codes.Internal, "Could not get finalized checkpoint: %v", err)
 	}
+	if cp == nil {
+		return nil, status.Errorf(codes.NotFound, "Finality not yet achieved")
+	}
```

Comment:
No test for this code as it's impossible to save a state with empty checkpoint in the DB.

Reply:
So this would be an impossible condition ? Why add a nil check here in that case.